git-conventional-commits: init at 2.7.2

yzx9 opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [x] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

yzx9 committed:
git-conventional-commits: init at 2.7.2

yzx9 commented:
git-conventional-commits

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 373930`

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>git-conventional-commits</li>
  </ul>
</details>

### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>git-conventional-commits</li>
  </ul>
</details>

donovanglover commented on pkgs/by-name/gi/git-conventional-commits/package.nix:
```diff
@@ -0,0 +1,46 @@
+
+  meta = with lib; {
+    homepage = "https://github.com/qoomon/git-conventional-commits";
+    description = "Git Conventional Commits Util to generate Semantic Version and Markdown Change Log and Validate Commit Message";
```
```suggestion
    description = "Generate semantic versions, markdown changelogs, and validate commit messages";
```

yzx9 replied:
Thank you! I have adopted your suggestion. However, I’m not sure about the style of description we should follow. Could you provide some guidance?

donovanglover replied:
See https://github.com/NixOS/nixpkgs/tree/master/pkgs#meta-attributes

yzx9 replied:
@donovanglover tks! May I request your review again?

donovanglover commented on pkgs/by-name/gi/git-conventional-commits/package.nix:
```diff
@@ -0,0 +1,46 @@
+  src = fetchFromGitHub {
+    owner = "qoomon";
+    repo = pname;
+    rev = "v${version}";
```
```suggestion
    tag = "v${version}";
```

donovanglover reviewed (commented):
Everything besides `installPhase` looks great. I haven't tested functionality but seems like the only difference after removing it is that the upstream CLI logic uses the file name instead of `git-conventional-commits`.

yzx9 commented on pkgs/by-name/gi/git-conventional-commits/package.nix:
```diff
@@ -0,0 +1,47 @@
+    mkdir -p $out/{bin,share/git-conventional-commits}
+    cp cli.js git-conventional-commits.default.yaml commit-msg.sh $out/share/git-conventional-commits
+    cp -r lib/ node_modules/ $out/share/git-conventional-commits
+    ln -s $out/share/git-conventional-commits/cli.js $out/bin/git-conventional-commits
```
You are right. it still works after removing `installPhase`. I really appreciate all the work done by the toolchain developers!

donovanglover commented on pkgs/by-name/gi/git-conventional-commits/package.nix:
```diff
@@ -0,0 +1,36 @@
+
+  meta = {
+    homepage = "https://github.com/qoomon/git-conventional-commits";
+    description = "Generate semantic version, markdown changelogs, and validate commit messages";
```
Nit: This isn't grammatically correct. Any ideas for a better description?

yzx9 replied:
How about changing **version** to the plural **versions** to align with the enumeration?

donovanglover replied:
SGTM

yzx9 commented:
I have updated the `meta.description` with correct grammar and added a simple test using `testers.testVersion`. @donovanglover, could you review it again?

ethancedwards8 commented on pkgs/by-name/gi/git-conventional-commits/package.nix:
```diff
@@ -0,0 +1,44 @@
+
+  dontNpmBuild = true;
+
+  passthru = {
+    tests.version = testers.testVersion {
+      package = git-conventional-commits;
+      command = "git-conventional-commits --version";
+    };
```
could this instead just be a versionCheckHook?

yzx9 replied:
Got it!

ethancedwards8 commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 373930`

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>git-conventional-commits</li>
  </ul>
</details>

ethancedwards8 reviewed (approved):
LGTM.

yzx9 commented on pkgs/by-name/gi/git-conventional-commits/package.nix:
```diff
@@ -0,0 +1,42 @@
+  buildNpmPackage,
+  fetchFromGitHub,
+  nix-update-script,
+  versionCheckHook,
```
done

yzx9 commented on pkgs/by-name/gi/git-conventional-commits/package.nix:
```diff
@@ -0,0 +1,42 @@
+  meta = {
+    homepage = "https://github.com/qoomon/git-conventional-commits";
+    description = "Generate semantic versions, markdown changelogs, and validate commit messages";
+    changelog = "https://github.com/qoomon/git-conventional-commits/blob/master/CHANGELOG.md";
```
got it

GaetanLepage reviewed (approved):
Good job @yzx9!
